src/lib/io/include/sol-uart.h

```diff
@@ -81,22 +89,27 @@ enum sol_uart_parity {
  * @brief Amount of stop bits.
  */
 enum sol_uart_stop_bits {
-    SOL_UART_STOP_BITS_ONE = 0,
-    SOL_UART_STOP_BITS_TWO
+    SOL_UART_STOP_BITS_ONE = 0, /**< Use one stop bit*/
+    SOL_UART_STOP_BITS_TWO /**< Use two stop bits */
 };
 
+/**
+ * @brief A configuration struct used to set the UART paramenters.
+ *
+ * @see sol_uart_open()
+ */
 struct sol_uart_config {
 #ifndef SOL_NO_API_VERSION
 #define SOL_UART_CONFIG_API_VERSION (1)
-    uint16_t api_version;
+    uint16_t api_version; /**< The API version */
 #endif
-    enum sol_uart_baud_rate baud_rate;
-    enum sol_uart_data_bits data_bits;
-    enum sol_uart_parity parity;
-    enum sol_uart_stop_bits stop_bits;
-    void (*rx_cb)(void *user_data, struct sol_uart *uart, uint8_t byte_read); /** Set a callback to be called every time a character is received on UART */
-    const void *rx_cb_user_data;
-    bool flow_control; /** Enables software flow control(XOFF and XON) */
+    enum sol_uart_baud_rate baud_rate; /**< The baud rate value */
+    enum sol_uart_data_bits data_bits; /**< The data bits value */
+    enum sol_uart_parity parity; /**< The parity value*/
+    enum sol_uart_stop_bits stop_bits; /**< The stop bits value */
+    void (*rx_cb)(void *user_data, struct sol_uart *uart); /** Set a callback to be called every time a character is received on UART */
+    const void *rx_cb_user_data; /**< User data to @c rx_cb */
```

Comment:
you can use simply `user_data`, if more callbacks are added they can use the same data.